src/snowflake/cli/_plugins/nativeapp/sf_sql_facade.py

```diff
@@ -206,3 +218,81 @@ def get_account_event_table(self, role: str | None = None) -> str | None:
         if table is None or table == "NONE":
             return None
         return table
+
+    def get_event_definitions(
+        self, app_name: str, role: str | None = None
+    ) -> list[dict]:
+        """
+        Retrieves event definitions for the specified application.
+        @param app_name: Name of the application to get event definitions for.
+        @return: A list of dictionaries containing event definitions.
+        """
+        query = (
+            f"show telemetry event definitions in application {to_identifier(app_name)}"
+        )
+        with self._use_role_optional(role):
+            try:
+                results = self._sql_executor.execute_query(
+                    query, cursor_class=DictCursor
+                ).fetchall()
+            except Exception as err:
+                handle_unclassified_error(
+                    err, f"Failed to get event definitions for application {app_name}."
+                )
+        return [dict(row) for row in results]
+
+    def get_app_properties(
+        self, app_name: str, role: str | None = None
+    ) -> Dict[str, str]:
+        """
+        Retrieve the properties of the specified application.
+        @param app_name: Name of the application.
+        @return: A dictionary containing the properties of the application.
+        """
+
+        query = f"desc application {to_identifier(app_name)}"
+        with self._use_role_optional(role):
+            try:
+                results = self._sql_executor.execute_query(
+                    query, cursor_class=DictCursor
+                ).fetchall()
+            except Exception as err:
+                handle_unclassified_error(
+                    err, f"Failed to describe application {app_name}."
+                )
+        return {row["property"]: row["value"] for row in results}
+
+    def get_ui_parameter(self, parameter: UIParameter, default: Any) -> str:
+        """
+        Returns the value of a single UI parameter.
+        If the parameter is not found, the default value is returned.
+        """
+
+        ui_parameters = self.get_ui_parameters()
+        return ui_parameters.get(parameter, default)
+
+    @lru_cache()
```

Comment:
this will not work well since we create a new facade each time we use get_snowflake_facade()

Reply:
Updated to be singleton.

Reply:
Using a ContextVar, right? I don't think we do pure global singletons anymore

Reply:
updated